### PR TITLE
Faster glitch flagging

### PR DESCRIPTION
**Pull request by tanaybhandarkar**
Based on the discussions in issue #536, I have put in a catch to supply a decimation factor for the Fourier filtered signal if there are over 50,000 samples in the AxisManager. The decimated array will now be around 20,000 samples, which will speed up the `stats.iqr` step.

### CODE REVIEW

**Comment by msilvafe** [comment]
That's great, roughly how much speedup do you get when you still estimate the iqr reasonably?

File: sotodlib/flags.py [hunk 1]
```diff
@@ -95,7 +95,10 @@ def get_glitch_flags(
     )
     # get the threshods based on n_sig x nlev = n_sig x iqu x 0.741
     fvec = np.abs(fvec)
-    iqr_range = 0.741 * stats.iqr(fvec, axis=1)
+    if fvec.shape[1] > 50000:
+        ds = int(fvec.shape[1]/20000)
+    else: ds = 1
```
Comment (mhasself) [inline sotodlib/flags.py:100]: two lines please

**Review by mhasself — CHANGES_REQUESTED** [review]
This is fine except for I don't like "if a: b" and "else: c" on one line!

**Comment by tanaybhandarkar** [comment]
> That's great, roughly how much speedup do you get when you still estimate the iqr reasonably? 


Without this adjustment, on a 264000 sample AxisManager, it takes about 60-90s for `get_glitch_flags` to fully run. When adding this downsampling option, it cuts that down to 20s. 

Caveat: There's a lot of variation on the timing, even when running the function without changing anything. I've chalked this up to a simons1 computing quirk. I never saw a 5 min runtime, like you mentioned to me earlier- if you know which context file you were working with, I can test with that one, too.

**Review by mhasself — APPROVED** [review]
Looks good to me but let Max or Katie confirm before merging.

**Comment by msilvafe** [comment]
Looks good to me I'm happy for you to merge.

**Pull request merged**